workspaces/ui-v2/src/pages/diffs/ReviewEndpointDiffPage/DiffLinks.tsx

```diff
@@ -1,134 +1,41 @@
-import React, { useMemo } from 'react';
-import sortBy from 'lodash.sortby';
+import React from 'react';
 
-import {
-  IInterpretation,
-  IRequestBodyLocation,
-  IResponseBodyLocation,
-} from '<src>/lib/Interfaces';
+import { IInterpretation } from '<src>/lib/Interfaces';
 import { ICopyRender } from '../components/ICopyRender';
 import { makeStyles } from '@material-ui/styles';
 
 import { List, ListItem, ListSubheader } from '@material-ui/core';
 
-type Section = {
-  requestId?: string;
-  responseId?: string;
-  statusCode?: number;
-  contentType?: string;
-};
-
 export function DiffLinks({
   allDiffs,
-  setSelectedDiffHash,
+  setSelectedDiff,
 }: {
   allDiffs: IInterpretation[];
-  setSelectedDiffHash: (hash: string) => void;
+  setSelectedDiff: (index: number) => void;
 }) {
   const classes = useStyles();
-  const sections = useMemo<Section[]>(() => {
-    const sections: Section[] = [];
-    const inRequests: IRequestBodyLocation[] = sortBy(
-      allDiffs
-        .filter((i) => i.diffDescription?.location!.inRequest)
-        .map((i) => i.diffDescription?.location.inRequest!),
-      'contentType'
-    );
-
-    inRequests.forEach((req) => {
-      const alreadyAdded = sections.find(
-        (i) => i.requestId && i.requestId === req.requestId
-      );
-      if (!alreadyAdded) {
-        sections.push({
-          requestId: req.requestId,
-          contentType: req.contentType,
-        });
-      }
-    });
-
-    const inResponses: IResponseBodyLocation[] = sortBy(
-      allDiffs
-        .filter((i) => i.diffDescription?.location!.inResponse)
-        .map((i) => i.diffDescription?.location.inResponse!),
-      'statusCode'
-    );
-
-    inResponses.forEach((res) => {
-      const alreadyAdded = sections.find(
-        (i) => i.responseId && i.responseId === res.responseId
-      );
-      if (!alreadyAdded) {
-        sections.push({
-          responseId: res.responseId,
-          statusCode: res.statusCode,
-          contentType: res.contentType,
-        });
-      }
-    });
-    return sections;
-  }, [allDiffs]);
-
   return (
     <List>
-      {sections.map((section) => {
-        if (section.requestId) {
-          return (
-            <div key={section.requestId}>
+      {/* TODO figure out why diffDescription can be nullable */}
```

Comment:
Those are generated on the Typescript side, and probably _can_ be nullable because new types of diffs can be introduced for which it doesn't map a description yet? Like we're about the add new diff types for query params!

Reply:
Hmmm, that could be valid but I think diffDescription is nullable because of loading state - as for other diff types we might want a different way to model this